
<!doctype html>
<html lang="en" class="no-js">
  <head>
    
      <meta charset="utf-8">
      <meta name="viewport" content="width=device-width,initial-scale=1">
      
      
      
      
      <link rel="icon" href="../assets/images/favicon.png">
      <meta name="generator" content="mkdocs-1.1.2, mkdocs-material-7.1.3">
    
    
      
        <title>Supplemental Methods - PLT</title>
      
    
    
      <link rel="stylesheet" href="../assets/stylesheets/main.e35208c4.min.css">
      
        
        <link rel="stylesheet" href="../assets/stylesheets/palette.ef6f36e2.min.css">
        
      
    
    
    
      
        
        <link rel="preconnect" href="https://fonts.gstatic.com" crossorigin>
        <link rel="stylesheet" href="https://fonts.googleapis.com/css?family=Roboto:300,400,400i,700%7CRoboto+Mono&display=fallback">
        <style>:root{--md-text-font-family:"Roboto";--md-code-font-family:"Roboto Mono"}</style>
      
    
    
    
    
      
    
    
  </head>
  
  
    
    
      
    
    
    
    
    <body dir="ltr" data-md-color-scheme="default" data-md-color-primary="white" data-md-color-accent="pink">
  
    
    <script>function __prefix(e){return new URL("..",location).pathname+"."+e}function __get(e,t=localStorage){return JSON.parse(t.getItem(__prefix(e)))}</script>
    
      <script>var palette=__get("__palette");if(null!==palette&&"object"==typeof palette.color)for(var key in palette.color)document.body.setAttribute("data-md-color-"+key,palette.color[key])</script>
    
    <input class="md-toggle" data-md-toggle="drawer" type="checkbox" id="__drawer" autocomplete="off">
    <input class="md-toggle" data-md-toggle="search" type="checkbox" id="__search" autocomplete="off">
    <label class="md-overlay" for="__drawer"></label>
    <div data-md-component="skip">
      
        
        <a href="#supplemental-methods" class="md-skip">
          Skip to content
        </a>
      
    </div>
    <div data-md-component="announce">
      
    </div>
    
      <header class="md-header" data-md-component="header">
  <nav class="md-header__inner md-grid" aria-label="Header">
    <a href=".." title="PLT" class="md-header__button md-logo" aria-label="PLT" data-md-component="logo">
      
  
  <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M12 8a3 3 0 0 0 3-3 3 3 0 0 0-3-3 3 3 0 0 0-3 3 3 3 0 0 0 3 3m0 3.54C9.64 9.35 6.5 8 3 8v11c3.5 0 6.64 1.35 9 3.54 2.36-2.19 5.5-3.54 9-3.54V8c-3.5 0-6.64 1.35-9 3.54z"/></svg>

    </a>
    <label class="md-header__button md-icon" for="__drawer">
      <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M3 6h18v2H3V6m0 5h18v2H3v-2m0 5h18v2H3v-2z"/></svg>
    </label>
    <div class="md-header__title" data-md-component="header-title">
      <div class="md-header__ellipsis">
        <div class="md-header__topic">
          <span class="md-ellipsis">
            PLT
          </span>
        </div>
        <div class="md-header__topic" data-md-component="header-topic">
          <span class="md-ellipsis">
            
              Supplemental Methods
            
          </span>
        </div>
      </div>
    </div>
    
      <form class="md-header__option" data-md-component="palette">
        
          
          
          <input class="md-option" data-md-color-media="" data-md-color-scheme="default" data-md-color-primary="white" data-md-color-accent="pink" type="radio" name="__palette" id="__palette_1">
          
            <label class="md-header__button md-icon" title="Switch to dark mode" for="__palette_2" hidden>
              <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M7 10a2 2 0 0 1 2 2 2 2 0 0 1-2 2 2 2 0 0 1-2-2 2 2 0 0 1 2-2m10-3a5 5 0 0 1 5 5 5 5 0 0 1-5 5H7a5 5 0 0 1-5-5 5 5 0 0 1 5-5h10M7 9a3 3 0 0 0-3 3 3 3 0 0 0 3 3h10a3 3 0 0 0 3-3 3 3 0 0 0-3-3H7z"/></svg>
            </label>
          
        
          
          
          <input class="md-option" data-md-color-media="" data-md-color-scheme="slate" data-md-color-primary="black" data-md-color-accent="pink" type="radio" name="__palette" id="__palette_2">
          
            <label class="md-header__button md-icon" title="Switch to light mode" for="__palette_1" hidden>
              <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M17 7H7a5 5 0 0 0-5 5 5 5 0 0 0 5 5h10a5 5 0 0 0 5-5 5 5 0 0 0-5-5m0 8a3 3 0 0 1-3-3 3 3 0 0 1 3-3 3 3 0 0 1 3 3 3 3 0 0 1-3 3z"/></svg>
            </label>
          
        
      </form>
    
    
    
      <label class="md-header__button md-icon" for="__search">
        <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M9.5 3A6.5 6.5 0 0 1 16 9.5c0 1.61-.59 3.09-1.56 4.23l.27.27h.79l5 5-1.5 1.5-5-5v-.79l-.27-.27A6.516 6.516 0 0 1 9.5 16 6.5 6.5 0 0 1 3 9.5 6.5 6.5 0 0 1 9.5 3m0 2C7 5 5 7 5 9.5S7 14 9.5 14 14 12 14 9.5 12 5 9.5 5z"/></svg>
      </label>
      
<div class="md-search" data-md-component="search" role="dialog">
  <label class="md-search__overlay" for="__search"></label>
  <div class="md-search__inner" role="search">
    <form class="md-search__form" name="search">
      <input type="text" class="md-search__input" name="query" aria-label="Search" placeholder="Search" autocapitalize="off" autocorrect="off" autocomplete="off" spellcheck="false" data-md-component="search-query" data-md-state="active" required>
      <label class="md-search__icon md-icon" for="__search">
        <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M9.5 3A6.5 6.5 0 0 1 16 9.5c0 1.61-.59 3.09-1.56 4.23l.27.27h.79l5 5-1.5 1.5-5-5v-.79l-.27-.27A6.516 6.516 0 0 1 9.5 16 6.5 6.5 0 0 1 3 9.5 6.5 6.5 0 0 1 9.5 3m0 2C7 5 5 7 5 9.5S7 14 9.5 14 14 12 14 9.5 12 5 9.5 5z"/></svg>
        <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M20 11v2H8l5.5 5.5-1.42 1.42L4.16 12l7.92-7.92L13.5 5.5 8 11h12z"/></svg>
      </label>
      <button type="reset" class="md-search__icon md-icon" aria-label="Clear" tabindex="-1">
        <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M19 6.41 17.59 5 12 10.59 6.41 5 5 6.41 10.59 12 5 17.59 6.41 19 12 13.41 17.59 19 19 17.59 13.41 12 19 6.41z"/></svg>
      </button>
    </form>
    <div class="md-search__output">
      <div class="md-search__scrollwrap" data-md-scrollfix>
        <div class="md-search-result" data-md-component="search-result">
          <div class="md-search-result__meta">
            Initializing search
          </div>
          <ol class="md-search-result__list"></ol>
        </div>
      </div>
    </div>
  </div>
</div>
    
    
  </nav>
</header>
    
    <div class="md-container" data-md-component="container">
      
      
        
      
      <main class="md-main" data-md-component="main">
        <div class="md-main__inner md-grid">
          
            
              
              <div class="md-sidebar md-sidebar--primary" data-md-component="sidebar" data-md-type="navigation" >
                <div class="md-sidebar__scrollwrap">
                  <div class="md-sidebar__inner">
                    


<nav class="md-nav md-nav--primary" aria-label="Navigation" data-md-level="0">
  <label class="md-nav__title" for="__drawer">
    <a href=".." title="PLT" class="md-nav__button md-logo" aria-label="PLT" data-md-component="logo">
      
  
  <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M12 8a3 3 0 0 0 3-3 3 3 0 0 0-3-3 3 3 0 0 0-3 3 3 3 0 0 0 3 3m0 3.54C9.64 9.35 6.5 8 3 8v11c3.5 0 6.64 1.35 9 3.54 2.36-2.19 5.5-3.54 9-3.54V8c-3.5 0-6.64 1.35-9 3.54z"/></svg>

    </a>
    PLT
  </label>
  
  <ul class="md-nav__list" data-md-scrollfix>
    
      
      
      

  
  
  
    <li class="md-nav__item">
      <a href="../index.html" class="md-nav__link">
        Pleiotropy Lineage Tracking (PLT) Project
      </a>
    </li>
  

    
      
      
      

  
  
  
    
    <li class="md-nav__item md-nav__item--nested">
      
      
        <input class="md-nav__toggle md-toggle" data-md-toggle="__nav_2" type="checkbox" id="__nav_2" >
      
      <label class="md-nav__link" for="__nav_2">
        Results
        <span class="md-nav__icon md-icon"></span>
      </label>
      <nav class="md-nav" aria-label="Results" data-md-level="1">
        <label class="md-nav__title" for="__nav_2">
          <span class="md-nav__icon md-icon"></span>
          Results
        </label>
        <ul class="md-nav__list" data-md-scrollfix>
          
            
  
  
  
    
    <li class="md-nav__item md-nav__item--nested">
      
      
        <input class="md-nav__toggle md-toggle" data-md-toggle="__nav_2_1" type="checkbox" id="__nav_2_1" >
      
      <label class="md-nav__link" for="__nav_2_1">
        Fitness patterns
        <span class="md-nav__icon md-icon"></span>
      </label>
      <nav class="md-nav" aria-label="Fitness patterns" data-md-level="2">
        <label class="md-nav__title" for="__nav_2_1">
          <span class="md-nav__icon md-icon"></span>
          Fitness patterns
        </label>
        <ul class="md-nav__list" data-md-scrollfix>
          
            
  
  
  
    <li class="md-nav__item">
      <a href="../Results/Fitness%20patterns/barcode-level.html" class="md-nav__link">
        Barcode-level
      </a>
    </li>
  

          
            
  
  
  
    <li class="md-nav__item">
      <a href="../Results/Fitness%20patterns/cluster-level.html" class="md-nav__link">
        Cluster-level
      </a>
    </li>
  

          
            
  
  
  
    <li class="md-nav__item">
      <a href="../Results/Fitness%20patterns/scatterplots.html" class="md-nav__link">
        Home v away comparisons
      </a>
    </li>
  

          
        </ul>
      </nav>
    </li>
  

          
            
  
  
  
    
    <li class="md-nav__item md-nav__item--nested">
      
      
        <input class="md-nav__toggle md-toggle" data-md-toggle="__nav_2_2" type="checkbox" id="__nav_2_2" >
      
      <label class="md-nav__link" for="__nav_2_2">
        Lineage summaries
        <span class="md-nav__icon md-icon"></span>
      </label>
      <nav class="md-nav" aria-label="Lineage summaries" data-md-level="2">
        <label class="md-nav__title" for="__nav_2_2">
          <span class="md-nav__icon md-icon"></span>
          Lineage summaries
        </label>
        <ul class="md-nav__list" data-md-scrollfix>
          
            
  
  
  
    <li class="md-nav__item">
      <a href="../Results/Lineage%20summaries/summaries.html" class="md-nav__link">
        Summary of adapted lineages by ploidy
      </a>
    </li>
  

          
        </ul>
      </nav>
    </li>
  

          
        </ul>
      </nav>
    </li>
  

    
      
      
      

  
  
    
  
  
    
    <li class="md-nav__item md-nav__item--active md-nav__item--nested">
      
      
        <input class="md-nav__toggle md-toggle" data-md-toggle="__nav_3" type="checkbox" id="__nav_3" checked>
      
      <label class="md-nav__link" for="__nav_3">
        Supplemental Information
        <span class="md-nav__icon md-icon"></span>
      </label>
      <nav class="md-nav" aria-label="Supplemental Information" data-md-level="1">
        <label class="md-nav__title" for="__nav_3">
          <span class="md-nav__icon md-icon"></span>
          Supplemental Information
        </label>
        <ul class="md-nav__list" data-md-scrollfix>
          
            
  
  
    
  
  
    <li class="md-nav__item md-nav__item--active">
      
      <input class="md-nav__toggle md-toggle" data-md-toggle="toc" type="checkbox" id="__toc">
      
      
        
      
      
        <label class="md-nav__link md-nav__link--active" for="__toc">
          Supplemental Methods
          <span class="md-nav__icon md-icon"></span>
        </label>
      
      <a href="si_methods.html" class="md-nav__link md-nav__link--active">
        Supplemental Methods
      </a>
      
        
<nav class="md-nav md-nav--secondary" aria-label="Table of contents">
  
  
  
    
  
  
    <label class="md-nav__title" for="__toc">
      <span class="md-nav__icon md-icon"></span>
      Table of contents
    </label>
    <ul class="md-nav__list" data-md-component="toc" data-md-scrollfix>
      
        <li class="md-nav__item">
  <a href="#double-barcode-design" class="md-nav__link">
    Double barcode design
  </a>
  
    <nav class="md-nav" aria-label="Double barcode design">
      <ul class="md-nav__list">
        
          <li class="md-nav__item">
  <a href="#overview" class="md-nav__link">
    Overview
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#ancestral-strain-construction" class="md-nav__link">
    Ancestral strain construction
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#construction-of-high-diversity-libraries" class="md-nav__link">
    Construction of high diversity libraries
  </a>
  
</li>
        
      </ul>
    </nav>
  
</li>
      
        <li class="md-nav__item">
  <a href="#experimental-evolution-protocols" class="md-nav__link">
    Experimental evolution protocols
  </a>
  
    <nav class="md-nav" aria-label="Experimental evolution protocols">
      <ul class="md-nav__list">
        
          <li class="md-nav__item">
  <a href="#growth-conditions-and-transfer-regime" class="md-nav__link">
    Growth conditions and transfer regime
  </a>
  
</li>
        
      </ul>
    </nav>
  
</li>
      
        <li class="md-nav__item">
  <a href="#bulk-fitness-assay" class="md-nav__link">
    Bulk Fitness Assay
  </a>
  
    <nav class="md-nav" aria-label="Bulk Fitness Assay">
      <ul class="md-nav__list">
        
          <li class="md-nav__item">
  <a href="#assembling-clone-pools" class="md-nav__link">
    Assembling clone pools
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#bulk-competition-protocols" class="md-nav__link">
    Bulk competition protocols
  </a>
  
</li>
        
      </ul>
    </nav>
  
</li>
      
        <li class="md-nav__item">
  <a href="#high-throughput-barcode-sequencing" class="md-nav__link">
    High-throughput barcode sequencing
  </a>
  
    <nav class="md-nav" aria-label="High-throughput barcode sequencing">
      <ul class="md-nav__list">
        
          <li class="md-nav__item">
  <a href="#dna-extraction" class="md-nav__link">
    DNA extraction
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#pcr-amplification" class="md-nav__link">
    PCR Amplification
  </a>
  
</li>
        
      </ul>
    </nav>
  
</li>
      
        <li class="md-nav__item">
  <a href="#fitness-inference" class="md-nav__link">
    Fitness inference
  </a>
  
    <nav class="md-nav" aria-label="Fitness inference">
      <ul class="md-nav__list">
        
          <li class="md-nav__item">
  <a href="#demultiplexing-illumina-reads-and-counting-barcodes" class="md-nav__link">
    Demultiplexing Illumina reads and counting barcodes
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#clustering-error-correcting" class="md-nav__link">
    Clustering / error correcting
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#final-steps" class="md-nav__link">
    Final steps
  </a>
  
</li>
        
      </ul>
    </nav>
  
</li>
      
        <li class="md-nav__item">
  <a href="#whole-genome-clone-sequencing" class="md-nav__link">
    Whole-genome clone sequencing
  </a>
  
    <nav class="md-nav" aria-label="Whole-genome clone sequencing">
      <ul class="md-nav__list">
        
          <li class="md-nav__item">
  <a href="#clone-selection" class="md-nav__link">
    Clone selection
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#library-preparation" class="md-nav__link">
    Library preparation
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#variant-calling-pipeline" class="md-nav__link">
    Variant calling pipeline
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#fastq-processing" class="md-nav__link">
    FASTQ processing
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#snp-and-small-indel-variant-calling" class="md-nav__link">
    SNP and small indel variant calling
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#structural-variant-calling" class="md-nav__link">
    Structural variant calling
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#variant-annotation" class="md-nav__link">
    Variant annotation
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#filtering-snps-small-indels-and-structural-variants" class="md-nav__link">
    Filtering SNPs, small indels, and structural variants
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#identifying-large-rearrangements-and-aneuploidies" class="md-nav__link">
    Identifying large rearrangements and aneuploidies
  </a>
  
</li>
        
      </ul>
    </nav>
  
</li>
      
        <li class="md-nav__item">
  <a href="#statistical-analyses-of-fitness-and-pleiotropy" class="md-nav__link">
    Statistical analyses of fitness and pleiotropy
  </a>
  
    <nav class="md-nav" aria-label="Statistical analyses of fitness and pleiotropy">
      <ul class="md-nav__list">
        
          <li class="md-nav__item">
  <a href="#statistical-analysis-of-fitness-patterns" class="md-nav__link">
    Statistical analysis of fitness patterns
  </a>
  
    <nav class="md-nav" aria-label="Statistical analysis of fitness patterns">
      <ul class="md-nav__list">
        
          <li class="md-nav__item">
  <a href="#identifying-adapted-lineages" class="md-nav__link">
    Identifying adapted lineages
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#identifying-auto-diploidized-lineages" class="md-nav__link">
    Identifying auto-diploidized lineages
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#clustering-and-variance-partitioning" class="md-nav__link">
    Clustering and variance partitioning
  </a>
  
</li>
        
      </ul>
    </nav>
  
</li>
        
          <li class="md-nav__item">
  <a href="#linking-phenotypes-with-genotypes" class="md-nav__link">
    Linking phenotypes with genotypes
  </a>
  
    <nav class="md-nav" aria-label="Linking phenotypes with genotypes">
      <ul class="md-nav__list">
        
          <li class="md-nav__item">
  <a href="#mutual-information" class="md-nav__link">
    Mutual information
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#hierarchical-gene-pathway-analysis" class="md-nav__link">
    Hierarchical gene--pathway analysis
  </a>
  
</li>
        
      </ul>
    </nav>
  
</li>
        
      </ul>
    </nav>
  
</li>
      
        <li class="md-nav__item">
  <a href="#references" class="md-nav__link">
    References
  </a>
  
</li>
      
    </ul>
  
</nav>
      
    </li>
  

          
        </ul>
      </nav>
    </li>
  

    
  </ul>
</nav>
                  </div>
                </div>
              </div>
            
            
              
              <div class="md-sidebar md-sidebar--secondary" data-md-component="sidebar" data-md-type="toc" >
                <div class="md-sidebar__scrollwrap">
                  <div class="md-sidebar__inner">
                    
<nav class="md-nav md-nav--secondary" aria-label="Table of contents">
  
  
  
    
  
  
    <label class="md-nav__title" for="__toc">
      <span class="md-nav__icon md-icon"></span>
      Table of contents
    </label>
    <ul class="md-nav__list" data-md-component="toc" data-md-scrollfix>
      
        <li class="md-nav__item">
  <a href="#double-barcode-design" class="md-nav__link">
    Double barcode design
  </a>
  
    <nav class="md-nav" aria-label="Double barcode design">
      <ul class="md-nav__list">
        
          <li class="md-nav__item">
  <a href="#overview" class="md-nav__link">
    Overview
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#ancestral-strain-construction" class="md-nav__link">
    Ancestral strain construction
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#construction-of-high-diversity-libraries" class="md-nav__link">
    Construction of high diversity libraries
  </a>
  
</li>
        
      </ul>
    </nav>
  
</li>
      
        <li class="md-nav__item">
  <a href="#experimental-evolution-protocols" class="md-nav__link">
    Experimental evolution protocols
  </a>
  
    <nav class="md-nav" aria-label="Experimental evolution protocols">
      <ul class="md-nav__list">
        
          <li class="md-nav__item">
  <a href="#growth-conditions-and-transfer-regime" class="md-nav__link">
    Growth conditions and transfer regime
  </a>
  
</li>
        
      </ul>
    </nav>
  
</li>
      
        <li class="md-nav__item">
  <a href="#bulk-fitness-assay" class="md-nav__link">
    Bulk Fitness Assay
  </a>
  
    <nav class="md-nav" aria-label="Bulk Fitness Assay">
      <ul class="md-nav__list">
        
          <li class="md-nav__item">
  <a href="#assembling-clone-pools" class="md-nav__link">
    Assembling clone pools
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#bulk-competition-protocols" class="md-nav__link">
    Bulk competition protocols
  </a>
  
</li>
        
      </ul>
    </nav>
  
</li>
      
        <li class="md-nav__item">
  <a href="#high-throughput-barcode-sequencing" class="md-nav__link">
    High-throughput barcode sequencing
  </a>
  
    <nav class="md-nav" aria-label="High-throughput barcode sequencing">
      <ul class="md-nav__list">
        
          <li class="md-nav__item">
  <a href="#dna-extraction" class="md-nav__link">
    DNA extraction
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#pcr-amplification" class="md-nav__link">
    PCR Amplification
  </a>
  
</li>
        
      </ul>
    </nav>
  
</li>
      
        <li class="md-nav__item">
  <a href="#fitness-inference" class="md-nav__link">
    Fitness inference
  </a>
  
    <nav class="md-nav" aria-label="Fitness inference">
      <ul class="md-nav__list">
        
          <li class="md-nav__item">
  <a href="#demultiplexing-illumina-reads-and-counting-barcodes" class="md-nav__link">
    Demultiplexing Illumina reads and counting barcodes
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#clustering-error-correcting" class="md-nav__link">
    Clustering / error correcting
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#final-steps" class="md-nav__link">
    Final steps
  </a>
  
</li>
        
      </ul>
    </nav>
  
</li>
      
        <li class="md-nav__item">
  <a href="#whole-genome-clone-sequencing" class="md-nav__link">
    Whole-genome clone sequencing
  </a>
  
    <nav class="md-nav" aria-label="Whole-genome clone sequencing">
      <ul class="md-nav__list">
        
          <li class="md-nav__item">
  <a href="#clone-selection" class="md-nav__link">
    Clone selection
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#library-preparation" class="md-nav__link">
    Library preparation
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#variant-calling-pipeline" class="md-nav__link">
    Variant calling pipeline
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#fastq-processing" class="md-nav__link">
    FASTQ processing
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#snp-and-small-indel-variant-calling" class="md-nav__link">
    SNP and small indel variant calling
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#structural-variant-calling" class="md-nav__link">
    Structural variant calling
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#variant-annotation" class="md-nav__link">
    Variant annotation
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#filtering-snps-small-indels-and-structural-variants" class="md-nav__link">
    Filtering SNPs, small indels, and structural variants
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#identifying-large-rearrangements-and-aneuploidies" class="md-nav__link">
    Identifying large rearrangements and aneuploidies
  </a>
  
</li>
        
      </ul>
    </nav>
  
</li>
      
        <li class="md-nav__item">
  <a href="#statistical-analyses-of-fitness-and-pleiotropy" class="md-nav__link">
    Statistical analyses of fitness and pleiotropy
  </a>
  
    <nav class="md-nav" aria-label="Statistical analyses of fitness and pleiotropy">
      <ul class="md-nav__list">
        
          <li class="md-nav__item">
  <a href="#statistical-analysis-of-fitness-patterns" class="md-nav__link">
    Statistical analysis of fitness patterns
  </a>
  
    <nav class="md-nav" aria-label="Statistical analysis of fitness patterns">
      <ul class="md-nav__list">
        
          <li class="md-nav__item">
  <a href="#identifying-adapted-lineages" class="md-nav__link">
    Identifying adapted lineages
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#identifying-auto-diploidized-lineages" class="md-nav__link">
    Identifying auto-diploidized lineages
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#clustering-and-variance-partitioning" class="md-nav__link">
    Clustering and variance partitioning
  </a>
  
</li>
        
      </ul>
    </nav>
  
</li>
        
          <li class="md-nav__item">
  <a href="#linking-phenotypes-with-genotypes" class="md-nav__link">
    Linking phenotypes with genotypes
  </a>
  
    <nav class="md-nav" aria-label="Linking phenotypes with genotypes">
      <ul class="md-nav__list">
        
          <li class="md-nav__item">
  <a href="#mutual-information" class="md-nav__link">
    Mutual information
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#hierarchical-gene-pathway-analysis" class="md-nav__link">
    Hierarchical gene--pathway analysis
  </a>
  
</li>
        
      </ul>
    </nav>
  
</li>
        
      </ul>
    </nav>
  
</li>
      
        <li class="md-nav__item">
  <a href="#references" class="md-nav__link">
    References
  </a>
  
</li>
      
    </ul>
  
</nav>
                  </div>
                </div>
              </div>
            
          
          <div class="md-content" data-md-component="content">
            <article class="md-content__inner md-typeset">
              
                
                
                <h1 id="supplemental-methods">Supplemental Methods</h1>
<h2 id="double-barcode-design">Double barcode design</h2>
<h3 id="overview">Overview</h3>
<p>The DNA barcoding system developed in Levy et al. [-@Levy15a] allows for the use of two distinct DNA barcodes – one in the landing pad itself, introduced via homologous recombination, and a second, which is introduced via <em>Cre-lox</em> mediated recombination. In Levy et al. [-@Levy15a], only a single sequence was used for the landing pad barcode, but here we used the landing pad barcode as a low complexity barcode to encode the experimental condition under which a population was evolved, while we used the other barcode (as in Levy et al. [-@Levy15a]) for lineage tracking itself. Figure S1 details the stepwise approach by which the barcoded populations were generated.</p>
<h3 id="ancestral-strain-construction">Ancestral strain construction</h3>
<p>To generate the ancestral strain for barcoding, we first constructed strain HR206, which contains the Magic Marker (Tong et al. 2004) by <strong>XXX</strong>. <em>HR206</em> was then crossed strain <em>SHA321</em> [@Jaffe17a], which carries the pre-landing pad, <em>Gal-Cre-NatMX</em>, at the <em>YBR209W</em> locus [@Levy15a]. <span class="arithmatex">\(\text{Mat}\alpha\)</span> spores derived from this cross were grown on Nourseothricin (Nat), to select for the pre-landing pad locus, and then one of these was backcrossed to FY3 (Winston et al., 1995). This process was repeated 5 times, each time selecting for <span class="arithmatex">\(\text{Mat}\alpha\)</span> spores containing the landing pad. A single spore derived from the final backcross was used as the ancestor for generating haploid barcoded libraries; one more mating with FY3 was performed to obtain the diploid ancestor.</p>
<p>We next introduced the barcoded landing pad into the pre-landing pad locus by homologous recombination with NatMX (Figure S1). The landing pad contains a <em>lox66</em> site, a DNA barcode (which we refer to as BC1), an artificial intron, the 3’ half of <em>URA3</em>, and <em>HygMX</em>. We PCR amplified the fragment of interest from the plasmid library L001 (~75,000 barcodes), described in Jaffe et al. [-@Jaffe17a] and transformed to obtain a library of landing pad strains containing a BC1 barcode. The transformation protocol is as described in detail in Levy et al [-@Levy15a].</p>
<h3 id="construction-of-high-diversity-libraries">Construction of high diversity libraries</h3>
<p>We selected 101 barcoded landing pad haploid strains and 117 barcoded landing pad diploid strains, listed in Table S2 with their associated environment for evolution. Each individual barcoded landing pad was then transformed using the plasmid library pBAR3-L1 (~500,000 barcodes) described in Levy et al. [-@Levy15a]. This plasmid carries <em>lox71</em>, a DNA barcode (referred to as BC2), an artificial intron, the 5’ half of <em>URA3</em>, and <em>HygMX</em>. Transformants were selected onto SC +Gal –Ura, to allow expression of the Cre recombinase, which is under the <em>GAL1</em> promoter. The recombination between <em>lox66</em> and <em>lox71</em> is irreversible and brings the two barcodes in close proximity to form an intron within the complete and functional <em>URA3</em> gene. For each landing pad strain, we generated <span class="arithmatex">\(10^{4}-10^{5}\)</span> transformants. The plates were scraped, and transformants from each plate were stored separately in glycerol at -80°C.</p>
<p>To estimate barcode diversity in each of the single libraries, we performed high throughput sequencing of the barcode region following DNA extraction (described in detail below). Using this sequence data, we quantified the distribution of frequencies of all unique barcodes present in each single library. We then pooled non-overlapping sets of these single libraries in order to obtain 12 pools (per ploidy) estimated to contain ~500,000 unique lineages (combination of BC1 and BC2). The number of landing pad (BC1) strains per pool ranged from 4 to 16. The minimum Hamming distance between landing pad barcodes within any pool is 6 for the diploid pools and 7 for the haploid pools.</p>
<p><strong>Table S1.</strong> Primers and details on DBC construction, primers, etc.</p>
<blockquote>
<p>WIP</p>
</blockquote>
<p><strong>Table S2.</strong> with statistics on library size, diversity, from ancestral pools.</p>
<blockquote>
<p>WIP</p>
</blockquote>
<h2 id="experimental-evolution-protocols">Experimental evolution protocols</h2>
<h3 id="growth-conditions-and-transfer-regime">Growth conditions and transfer regime</h3>
<p>Each starting pool was evolved by serial batch culture in 500 mL baffled flasks in various conditions described below (Table S3). Unless otherwise specified, vegetative propagation was conducted in in 100 mL of each media at 30°C with shaking at 225 rpm.</p>
<p>Base medium for most conditions was SC complete (YNB+ Nitrogen- 1501-500, SC-1300-500, Sunrise Science Products) supplemented with 2% Glucose (BD-Difco-215510). The evolution experiments were started with a pre-culture of each pool grown in 100 mL of SC-2% Glucose at 30°C overnight. This pre-culture was used to inoculate both replicate evolutions with <span class="arithmatex">\(\sim 10^7\)</span> cells (~400 µL).</p>
<p><strong>Table S3.</strong> Media recipe per source environment.</p>
<blockquote>
<p>WIP</p>
</blockquote>
<p>Serial transfers were performed by transferring <span class="arithmatex">\(\sim 10^7\)</span> cells (~ 400 µL) into fresh media at the designated transfer time (24 hours for most environments; Table S3). We generated archival glycerol stocks (3 <span class="arithmatex">\(\times\)</span> 1 mL culture at 16.6% final glycerol concentration), and pelleted, washed, and concentrated the remaining culture volume (90 mL) for later DNA extraction and barcode sequencing (described below).</p>
<h2 id="bulk-fitness-assay">Bulk Fitness Assay</h2>
<h3 id="assembling-clone-pools">Assembling clone pools</h3>
<p>We generated pools of adapted genotypes by isolating individual yeast clones from each of our evolution environments. To do so, we first determined appropriate time point(s) from which to sample in order to gather a large number of unique lineages carrying distinct beneficial mutations.</p>
<p>When isolating lineages from time points throughout the evolutions, we were faced with trying to balance requiring that clones be adapted with the necessity of sampling as many <em>unique</em> adapted lineages at once. We first measured population mean fitness using bulk population fitness assays against a fluorescently labeled reference strain (describe below) to determine time points with high mean fitness. For many environments, sampling clones from time-point(s) shortly after mean fitness began to consistently increase yielded sufficiently high numbers of lineages that were adapted and had unique barcodes.</p>
<p>However, several evolution environments showing positive mean fitness were dominated by a very small number of lineages, such that few unique barcodes were recovered after clone isolation. To solve this problem, we took advantage of low-coverage barcode sequencing of various time points from the initial evolutions. Using a simple heuristic to define whether barcodes were adapted, we calculated the time point which would yield the largest number of unique and adapted barcodes in a sample of <span class="arithmatex">\(N\)</span> clones per time point.</p>
<p>To isolate clones from the selected time points, we inoculated a ~10 µL aliquot of frozen glycerol stocks of the target sample into 5 mL of YPD and grew this pre-culture overnight. This culture was suspended at <span class="arithmatex">\(10^{5}\)</span> cells per mL in PBS before clone isolation into 96-well plates by way of single cell sorting.</p>
<h3 id="bulk-competition-protocols">Bulk competition protocols</h3>
<blockquote>
<p>WIP: Parris</p>
</blockquote>
<h2 id="high-throughput-barcode-sequencing">High-throughput barcode sequencing</h2>
<h3 id="dna-extraction">DNA extraction</h3>
<p>From a dry cell pellet stored at -20°, DNA was extracted using the MasterPure™ Yeast DNA Purification Kit (Epicentre MPY80200) with slight modifications compared to the manufacturer’s guidelines: the lysis step was performed for one hour in lysis buffer, supplemented with RNAse at 1.66 µg/µL. Prior to resuspension with sterile water, each DNA pellet was washed twice with 70% Ethanol to remove remaining contaminants.</p>
<p>Following quantification via QuBit™, DNA was resuspended to a final concentration of 50 ng/µL.</p>
<h3 id="pcr-amplification">PCR Amplification</h3>
<p>Amplification of the barcode region was performed as previously described [@Levy15a; @Venkataram16a], with the following amendments.</p>
<p><strong>Table S4</strong>. Primers used for library construction</p>
<blockquote>
<p>WIP</p>
</blockquote>
<p><strong>Table S5.</strong> Time points and replicates used for fitness inference from all BFA assays.</p>
<blockquote>
<p>WIP</p>
</blockquote>
<h2 id="fitness-inference">Fitness inference</h2>
<h3 id="demultiplexing-illumina-reads-and-counting-barcodes">Demultiplexing Illumina reads and counting barcodes</h3>
<p>We first divide reads into individual libraries based on inline indices following the unique molecular index on the paired-end reads.  We discard reads if the average quality score of the barcode regions is less than 30.  The unique molecular index (UMI) for a set of paired reads is the first 8 bases of read 1 plus the first 8 bases of read 2.  For each library, we discard reads with duplicate UMIs. Finally, we extract barcodes by searching the barcode region, plus 10 bases on either side, using the regular expression below.  We discard reads that do not match this regular expression</p>
<p><code>('\D*?(GTACC|GGACC|GGTCC|G.TACC|GG.ACC|GGT.CC|GGTA.C|GGTAC.)(\D{24,28})(.TAACT|A.AACT|AT.ACT|ATA.CT|ATAA.T|ATAAC|AAACT|ATACT|ATAAT)\D*')</code></p>
<h3 id="clustering-error-correcting">Clustering / error correcting</h3>
<p>Because of PCR and sequencing errors, many of the barcodes counted in the previous step are simply errors and should not be counted as “true” barcodes.  To error correct the set of barcodes for each experiment, we cluster errors to true barcodes based on the edit distance between barcodes.  We use the following algorithm to cluster the environment barcodes and diverse barcodes separately.  The algorithm takes advantage of the fact that all errors should be connected by single insertions, deletions, and substitutions by using deletion neighborhoods to speed up the error-correction process. The algorithm uses as input a list of barcodes and total counts (reads) corresponding to each barcode.  The steps are listed below:</p>
<ul>
<li>Make deletion neighborhoods</li>
<li>For each barcode, create the set of single-base deletions at each position</li>
<li>Connect barcodes with overlapping single-base deletion sets
    <em>Note: this overlap indicates that the two barcodes are separated by one “edit”: a substitution, insertion, or deletion</em></li>
<li>
<p>Within each neighborhood, define peaks by these criteria:</p>
<ul>
<li>Barcode does not contain an uncalled base (“N”)</li>
<li>Barcode has no single-edit neighbors with more total counts</li>
<li>Barcode has more than 10 total counts</li>
<li>Barcode is more than 3 edits away from any peak with more total counts</li>
</ul>
</li>
<li>
<p>Within each neighborhood, error correct non-peak barcodes:</p>
<ul>
<li>Check the Levenshtein (edit) distance between each non-peak barcode and each peak barcode.  If the edit distance is less than or equal to 3, the barcode error-corrects to the peak barcode.  If a barcode is within 3 edits of more than one peak, it corrects to the barcode with higher total counts
<em>Note: This step uses the python Levenshtein module</em> (https://pypi.python.org/pypi/python-Levenshtein/0.12.0#license)</li>
</ul>
</li>
</ul>
<p>Once we have error-corrected both the environment and diverse barcodes, we define the combinations of “true” environment and diverse barcodes, which we call “centroids.” We discard combinations that have less than or equal to 10 total counts and add the counts from error barcodes to the appropriate centroids to produce a final list of barcodes and counts.</p>
<h3 id="final-steps">Final steps</h3>
<p>Because the bulk fitness assays were all sequenced over multiple lanes, we perform the above steps for each lane separately and then combine count files based on full barcode (environment barcode + diverse barcode) sequences.  We discard any full barcodes that are not found in all lanes, which removes any lane-specific sequencing contamination.  Chimeric barcode pairs, which arise from sequencing or PCR issues, occur at a low rate in our reads. We identify chimeras by finding barcodes that share a diverse barcode (but not an environment barcode) with another barcode that has at least 100X more reads.  These barcodes are removed from the dataset.</p>
<h2 id="whole-genome-clone-sequencing">Whole-genome clone sequencing</h2>
<h3 id="clone-selection">Clone selection</h3>
<blockquote>
<p>WIP: Parris/Milo</p>
</blockquote>
<p><strong>Table S7.</strong> Adapted clones per environment inferred from BFAs, with number with WGS sequencing data.</p>
<blockquote>
<p>WIP</p>
</blockquote>
<h3 id="library-preparation">Library preparation</h3>
<blockquote>
<p>WIP: Lucas/Parris</p>
</blockquote>
<h3 id="variant-calling-pipeline">Variant calling pipeline</h3>
<p><strong>Table S8.</strong> Number of mutations discovered per clone per source and ploidy.</p>
<blockquote>
<p>WIP</p>
</blockquote>
<p>The pipeline for variants calling has been detailed in Li, et al., 2019 (Nature Eco&amp;Evo). Briefly, Sentieon Genomic Tools Version 201711.02 (REF: <code>doi:10.1101/115717</code>) were used for SNP, small indel and structural variants calling with <em>S. cerevisiae</em> S288C reference genome R64-1-1 (https://downloads.yeastgenome.org/sequence/S288C_reference/genome_releases/). The source code for variants calling and annotation (section 7.2-7.6) can be found at https://github.com/liyuping927/DNAscope-variants-calling.</p>
<h3 id="fastq-processing">FASTQ processing</h3>
<p>For each sample, we received two <code>.fastq</code> files, one for each read of the paired end sequencing ('fastqR1' and 'fastqR2'). Using <code>cutadapt v.1.16</code>, we trimmed the first 10 bp of each read (<code>-u 10</code>), low-quality ends (<code>-q 30</code>) and any adapter sequences (<code>-a</code>). After trimming, sequences with a length shorter than 12 bp (<code>--minimum-length 12</code>) were discarded.</p>
<p>First we trimmed the forward read, writing output to temporary files (note, commands are a single line):</p>
<pre><code>cutadapt --minimum-length 12 -q 30 -u 10 -a CTGTCTCTTATACACATCTCCGAGCCCACGAGAC -o tmp.1.fastq.gz -p tmp.2.fastq.gz fastqR1 fastqR2
</code></pre>
<p>Next we trimmed the reverse read, using the temporary files as input:</p>
<pre><code>cutadapt --minimum-length 12 -q 30 -u 10 -a CTGTCTCTTATACACATCTGACGCTGCCGACGA -o trimmedR2.fastq.gz -p trimmedR1.fastq.gz tmp.2.fastq.gz tmp.1.fastq.gz
</code></pre>
<p>We then mapped reads using <code>bwa</code> (citation) to <em>S. cerevisiae</em> S288C reference genome R64-1-1 (<a href="https://downloads.yeastgenome.org/sequence/S288C_reference/genome_releases/">https://downloads.yeastgenome.org/sequence/S288C_reference/genome_releases/</a>) and sorted using <code>Sentieon Genomic Tools v.6</code>:</p>
<pre><code>bwa mem -M -R readGroupInfo -K 10000000 ReferenceGenome trimmedR1.fastq.gz trimmedR2.fastq.gz) | sentieon util sort -o SORTED_BAM --sam2bam -i
</code></pre>
<p>Duplicates were removed using the sorted .<code>bam</code> file. The first command collected read information, and the second command performed the de-duping:</p>
<pre><code>sentieon driver -i SORTED_BAM --algo LocusCollector --fun score_info SCORE_TXT
sentieon driver -i SORTED_BAM --algo Dedup -- rmdup --score_info SCORE_TXT --metrics DEDUP_METRIC_TXT DEDUP_BAM
</code></pre>
<p>Local realignment around indels was performed using the deduped <code>.bam</code> file:</p>
<pre><code>sentieon driver -r ReferenceGenome -i DEDUP_BAM -- algo Realigner REALIGNED_BAM
</code></pre>
<p>Lastly, base quality score re-calibration was performed using the realigned <code>.bam</code> file. The first command calculated the required modification of the quality scores assigned to individual read bases of the sequence read data. The second command applied the re-calibration to calculate the post calibration data table:</p>
<pre><code>sentieon driver -r ReferenceGenome -i REALIGNED_BAM --algo QualCal RECAL_DATA.TABLE
sentieon driver -r ReferenceGenome -i REALIGNED_BAM -q RECAL_DATA.TABLE --algo QualCal RECAL_DATA.TABLE.POST
</code></pre>
<h3 id="snp-and-small-indel-variant-calling">SNP and small indel variant calling</h3>
<p>SNP and small indels variants were called by the <code>DNAscope</code> algorithm (Sentieon Genomic Software) using the realigned <code>.bam</code> file and the output table of the base quality score recalibration (<code>RECAL_DATA.TABLE</code>). The parameter <code>ploidy</code> is assigned as 1 for haploids and as 2 for diploids.</p>
<pre><code>sentieon driver -r ReferenceGenome -i REALIGNED_BAM -q RECAL_DATA.TABLE --algo DNAscope --ploidy [1|2] VARIANT_VCF
</code></pre>
<h3 id="structural-variant-calling">Structural variant calling</h3>
<p>The first command enabled the <code>DNAscope</code> algorithm to detect the break-end variant type (BND). The parameter <code>ploidy</code> was assigned as 1 for haploids and as 2 for diploids. The second command processed the temporary <code>.vcf</code>file using the <code>SVSolver</code> algorithm and output structural variants to a <code>.vcf</code> file.</p>
<pre><code>sentieon driver -r ReferenceGenome -i REALIGNED_BAM -q RECAL_DATA.TABLE --algo DNAscope --var_type bnd --ploidy [1|2] TMP_VARIANT_VCF
</code></pre>
<p>Then process the <code>.vcf</code> using the <code>SVSolver</code> algorithm with the following command:</p>
<pre><code>sentieon driver -r ReferenceGenome --algo SVSolver -v TMP_VARIANT_VCF STRUCTURAL_VARIANT_VCF
</code></pre>
<h3 id="variant-annotation">Variant annotation</h3>
<p>Here the <code>.vcf</code> file from SNP and small indel variants calling (<code>VARIANT_VCF</code>) is used as an example. The same commands were used to annotate structural variants.</p>
<p>We then used <code>snpEff7</code> (<a href="http://snpeff.sourceforge.net/download.html">http://snpeff.sourceforge.net/download.html</a>) to annotate a <code>.vcf</code>file and output the annotated <code>.vcf</code>file, named <code>Ann.vcf</code>.</p>
<pre><code>java -Xmx2g -jar snpEff -c snpEff_config -v R64-1-1.75 -class VARIANT_VCF &gt; Ann.vcf -s snpEff_summary.html
</code></pre>
<p>For variants in coding regions, <code>SNPSift</code> was used to extract the first annotation of each variant, which is the annotation with the largest effect. Output the extracted annotation as a <code>.vcf</code> file, named <code>Final_Ann.vcf</code>:</p>
<pre><code>java -jar SnpSift extractFields Ann.vcf CHROM POS ID REF ALT QUAL FILTER EFF[0].EFFECT EFF[0].GENE: EFF[0].IMPACT: EFF[0].FUNCLASS: EFF[0].CODON: EFF[0].AA ANN[0].BIOTYPE: GEN[0].GT GEN[0].AD GEN[0].DP GEN[0].GQ GEN[0].PL &gt; Final_Ann.vcf
</code></pre>
<p>For variants in non-coding regions, the nearest gene of each variant was extracted. Thus, the non-coding variants were annotated as either the upstream or downstream of the nearest genes.</p>
<h3 id="filtering-snps-small-indels-and-structural-variants">Filtering SNPs, small indels, and structural variants</h3>
<p>First, mitochondrial variants were discarded. Second, any variants in genes <em>FLO1</em> and <em>FLO9</em> were filtered out due to poor alignment in both genomic regions. Third, diploids with an average coverage lower than 15 and haploids with an average coverage lower than 10 were discarded. Fourth, background variants were removed. If a variant is present in &gt;~12% of clones isolated from the same evolutionary condition, this variant is considered as a background variant and discarded. Fifth, variants with a quality score smaller than 150 were filtered out. Note that if a variant was present in multiple clones, the alignment of this variant was manually checked regardless of its quality score and a decision was made based on all clones carrying this variant. Thus, a variant with a quality score &lt;150 may not be filtered out if the same variant contained in other clones was proven to be authentic. Similarly, a variant with a quality score &gt;150 may be filtered out if the same variant was proven to be bogus in other clones.</p>
<p>Furthermore, all variants were further verified by manually checking <code>.bam</code> files after alignment. By doing this, variants within repetitive regions and regions with a poor alignment were filtered out. More importantly, due to mishandling, sequencing data used here is contaminated by other yeast species/isolates at a low frequency (0-30%). While this low frequency contamination does not pose a big problem for variants calling in haploids, it leads to excessive miscalling of heterozygous variants in diploids. These miscalled heterozygous variants are caused by genetic variations between S288C (the <em>S. cerevisiae</em> strain used in this study) and the contaminated yeast source and often appear in “patch” (multiple variants within a small region, e.g. 4 variants within 100 bp), which is statistically impossible considering the short-period evolutionary time. To remove false variants caused by contamination, we manually checked alignment through <code>.bam</code>files and removed heterozygous variants appearing in patch. In addition, the ratio of ref:alt is used to assist the removal of false variants. Variants with Ref:Alt &gt;3:1 are very likely to be a result of low-frequency contamination. Lastly, ambiguous variants were checked using blast. Variants that are not present in other yeast species/isolates were considered as <em>de novo</em> mutations arising during the course of evolution.</p>
<h3 id="identifying-large-rearrangements-and-aneuploidies">Identifying large rearrangements and aneuploidies</h3>
<blockquote>
<p>WIP: Lucas</p>
</blockquote>
<h2 id="statistical-analyses-of-fitness-and-pleiotropy">Statistical analyses of fitness and pleiotropy</h2>
<blockquote>
<p>WIP Parris</p>
</blockquote>
<h3 id="statistical-analysis-of-fitness-patterns">Statistical analysis of fitness patterns</h3>
<h4 id="identifying-adapted-lineages">Identifying adapted lineages</h4>
<h4 id="identifying-auto-diploidized-lineages">Identifying auto-diploidized lineages</h4>
<h4 id="clustering-and-variance-partitioning">Clustering and variance partitioning</h4>
<p>Table S8. Number of lineages in each of <span class="arithmatex">\(k\)</span> clusters defines for each source environment, as well as robustness statistics to groupings.</p>
<blockquote>
<p>WIP</p>
</blockquote>
<h3 id="linking-phenotypes-with-genotypes">Linking phenotypes with genotypes</h3>
<h4 id="mutual-information">Mutual information</h4>
<h4 id="hierarchical-gene-pathway-analysis">Hierarchical gene--pathway analysis</h4>
<hr>

<h2 id="references">References</h2>
                
              
              
                


              
            </article>
          </div>
        </div>
        
      </main>
      
        
<footer class="md-footer">
  
    <nav class="md-footer__inner md-grid" aria-label="Footer">
      
        <a href="../Results/Lineage%20summaries/summaries.html" class="md-footer__link md-footer__link--prev" rel="prev">
          <div class="md-footer__button md-icon">
            <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M20 11v2H8l5.5 5.5-1.42 1.42L4.16 12l7.92-7.92L13.5 5.5 8 11h12z"/></svg>
          </div>
          <div class="md-footer__title">
            <div class="md-ellipsis">
              <span class="md-footer__direction">
                Previous
              </span>
              Summary of adapted lineages by ploidy
            </div>
          </div>
        </a>
      
      
    </nav>
  
  <div class="md-footer-meta md-typeset">
    <div class="md-footer-meta__inner md-grid">
      <div class="md-footer-copyright">
        
        Made with
        <a href="https://squidfunk.github.io/mkdocs-material/" target="_blank" rel="noopener">
          Material for MkDocs
        </a>
        
      </div>
      
    </div>
  </div>
</footer>
      
    </div>
    <div class="md-dialog" data-md-component="dialog">
      <div class="md-dialog__inner md-typeset"></div>
    </div>
    <script id="__config" type="application/json">{"base": "..", "features": [], "translations": {"clipboard.copy": "Copy to clipboard", "clipboard.copied": "Copied to clipboard", "search.config.lang": "en", "search.config.pipeline": "trimmer, stopWordFilter", "search.config.separator": "[\\s\\-]+", "search.placeholder": "Search", "search.result.placeholder": "Type to start searching", "search.result.none": "No matching documents", "search.result.one": "1 matching document", "search.result.other": "# matching documents", "search.result.more.one": "1 more on this page", "search.result.more.other": "# more on this page", "search.result.term.missing": "Missing"}, "search": "../assets/javascripts/workers/search.fe42c31b.min.js", "version": null}</script>
    
    
      <script src="../assets/javascripts/bundle.4ea5477f.min.js"></script>
      
        <script src="https://cdnjs.cloudflare.com/ajax/libs/tablesort/5.2.1/tablesort.min.js"></script>
      
        <script src="../javascripts/tables.js"></script>
      
        <script src="../javascripts/config.js"></script>
      
        <script src="https://polyfill.io/v3/polyfill.min.js?features=es6"></script>
      
        <script src="https://cdn.jsdelivr.net/npm/mathjax@3/es5/tex-mml-chtml.js"></script>
      
    
  </body>
</html>